pkg/foghorn/controller_test.go

```diff
@@ -0,0 +1,157 @@
+package foghorn
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	lighthousev1alpha1 "github.com/jenkins-x/lighthouse/pkg/apis/lighthouse/v1alpha1"
+	"github.com/jenkins-x/lighthouse/pkg/config"
+	"github.com/jenkins-x/lighthouse/pkg/plugins"
+	"github.com/jenkins-x/lighthouse/pkg/util"
+	"github.com/jenkins-x/lighthouse/pkg/watcher"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/yaml"
+)
+
+func TestReconcile(t *testing.T) {
+	testCases := []string{
+		"status-change",
+		"no-status-change",
+	}
+
+	configAgent := &config.Agent{}
+	configAgent.Set(&config.Config{
+		JobConfig: config.JobConfig{},
+		ProwConfig: config.ProwConfig{
+			Keeper:                 config.Keeper{},
+			Plank:                  config.Plank{},
+			Sinker:                 config.Sinker{},
+			Deck:                   config.Deck{},
+			BranchProtection:       config.BranchProtection{},
+			Orgs:                   nil,
+			Gerrit:                 config.Gerrit{},
+			JenkinsOperators:       nil,
+			LighthouseJobNamespace: "",
+			PodNamespace:           "",
+			LogLevel:               "",
+			PushGateway:            config.PushGateway{},
+			OwnersDirExcludes:      nil,
+			OwnersDirBlacklist:     nil,
+			PubSubSubscriptions:    nil,
+			GitHubOptions:          config.GitHubOptions{},
+			ProviderConfig: &config.ProviderConfig{
+				Kind:    "fake",
+				Server:  "https://github.com",
+				BotUser: "jenkins-x-bot",
+			},
+		},
+	})
+
+	pluginAgent := &plugins.ConfigAgent{}
+	pluginAgent.Set(&plugins.Configuration{
+		Plugins:                    nil,
+		ExternalPlugins:            nil,
+		Owners:                     plugins.Owners{},
+		Approve:                    nil,
+		UseDeprecatedSelfApprove:   false,
+		UseDeprecatedReviewApprove: false,
+		Blockades:                  nil,
+		Cat:                        plugins.Cat{},
+		CherryPickUnapproved:       plugins.CherryPickUnapproved{},
+		ConfigUpdater:              plugins.ConfigUpdater{},
+		Heart:                      plugins.Heart{},
+		Label:                      plugins.Label{},
+		Lgtm:                       nil,
+		RepoMilestone:              nil,
+		RequireMatchingLabel:       nil,
+		RequireSIG:                 plugins.RequireSIG{},
+		SigMention:                 plugins.SigMention{},
+		Size:                       plugins.Size{},
+		Triggers:                   nil,
+		Welcome:                    nil,
+	})
+
+	cfgMapWatcher := &watcher.ConfigMapWatcher{}
+
+	ns := "jx"
+
+	for _, tc := range testCases {
+		t.Run(tc, func(t *testing.T) {
+			testData := path.Join("test_data", tc)
+			_, err := os.Stat(testData)
+			assert.NoError(t, err)
+
+			observedJob, err := loadLighthouseJob(true, testData)
```

Comment:
So why the boolean opposed to just pass "observed-lhjob.yml" vs "expected-lhjob.yml". Took me a bit to understand what's going on here.

Reply:
Because I copy-pasted and made it unnecessary complicated. =)